cmd/client-keystone-auth/main.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"k8s.io/client-go/pkg/apis/clientauthentication/v1alpha1"
+	"k8s.io/cloud-provider-openstack/pkg/identity/keystone"
+)
+
+// KuberneteExecInfo holds information passed to the plugin by the transport. This
+// contains runtime specific information, such as if the session is interactive,
+// auth API version and kind of request.
+type KuberneteExecInfo struct {
+	APIVersion string                      `json:"apiVersion"`
+	Kind       string                      `json:"kind"`
+	Spec       v1alpha1.ExecCredentialSpec `json:"spec"`
+}
+
+func main() {
+	keiData := os.Getenv("KUBERNETES_EXEC_INFO")
+	if keiData == "" {
+		fmt.Fprintln(os.Stderr, "KUBERNETES_EXEC_INFO env variable must be set.")
+		os.Exit(1)
+	}
+
+	kei := KuberneteExecInfo{}
+	json.Unmarshal([]byte(keiData), &kei)
+
+	if kei.APIVersion != "client.authentication.k8s.io/v1alpha1" {
```

Comment:
I'd prefer to have a separate validation method.

Also the errors can be aggregated.

Reply:
This kind hard coded version is not k8s style. I'd prefer to import the GVK from `k8s.io/client-go/pkg/apis/clientauthentication` instead.